contracts/validator-manager/Native721TokenStakingManager.sol

```diff
@@ -326,6 +330,29 @@ contract Native721TokenStakingManager is
         emit RewardRegistered(primary, epoch, token, amount);
     }
 
+    /**
+     * @notice See {INative721TokenStakingManager-registerPrimaryRewards}.
+     */
+    function registerPrimaryRewards() external payable virtual nonReentrant {
```

Comment:
we dont really need to change the contracts because this part can simply be handled outside. Just manually wrap the beam before sending it to these contracts, its a one step tx to be added in the scripts before the register rewards step
Also this would require re audit and adds a bit of code duplication